src/Model.js

```diff
@@ -761,7 +760,7 @@ class Model {
    * @return {Array}
    */
   get attributeNames () {
-    return Object.keys(this.attributeDefinitions)
+    return this.attributeDefinitions ? Object.keys(this.attributeDefinitions) : []
```

Comment:
If we can ensure that `this.attributeDefinitions` is always an object, even if empty, we shouldn't have to do this.

Reply:
Sure, just have to adjust it in the schema